src/switchroot/ostree-mount-util.h

```diff
@@ -106,6 +128,115 @@ read_proc_cmdline_ostree (void)
   return ret;
 }
 
+/* If the key matches the start of the line, strip the key from the line just
+ * to get the value
+ */
+static inline bool
+rm_key_if_match (char *line, const char *key, const size_t key_len, char *out)
+{
+  if (strstr (line, key) == line)
+    {
+      char *value = line + key_len;
+      strcpy (out, value);
+      return true;
+    }
+
+  return false;
+}
+
+/* On completion version and options will be set to new values if the version
+ * is more recent. Will loop line through line on the passed in open FILE.
+ */
+static inline int
+copy_if_higher_version (FILE *f, char *version, char *options)
+{
+  char __attribute__ ((cleanup (free_char))) *line = NULL;
+  char version_local[PATH_MAX] = "";
+  char options_local[PATH_MAX] = "";
+  /* Note getline() will reuse the previous buffer */
+  for (size_t len = 0; getline (&line, &len, f) != -1;)
+    {
+      /* This is an awful hack to avoid depending on GLib in the
+       * initramfs right now.
+       */
+      if (rm_key_if_match (line, "version ", sizeof ("version"), version_local))
+        {
+          if (options_local[0])
+            break;
+
+          continue;
+        }
+
+      if (rm_key_if_match (line, "options ", sizeof ("options"), options_local))
+        {
+          if (version_local[0])
+            break;
+
+          continue;
+        }
+    }
+
+  const int ret = strverscmp (version_local, version);
+  if (ret > 0)
+    {
+      strcpy (version, version_local);
+      strcpy (options, options_local);
+    }
+
+  return ret;
+}
+
+/* This function is for boot arrangements where it is not possible to use a
+ * karg/cmdline, this is the case when the cmdline is part of the signed
+ * boot image, alternatively this function takes the karg from the bls entry
+ * which will have the correct ostree= karg set. This bls entry is not parsed
+ * from the bootloader but from the initramfs instead.
+ */
+static inline char *
+bls_parser_get_ostree_option (const char *sysroot)
+{
+  char out[PATH_MAX] = "";
+  int written = snprintf (out, PATH_MAX, "%s/boot/loader/entries", sysroot);
+  DIR __attribute__ ((cleanup (close_dir))) *dir = opendir (out);
+  if (!dir)
+    {
+      fprintf (stderr, "opendir(\"%s\") failed with %d\n", out, errno);
+      return strdup ("");
+    }
+
+  char version[PATH_MAX] = "";
+  char options[PATH_MAX] = "";
+  for (struct dirent *ent = 0; (ent = readdir (dir));)
+    {
+      if (ent->d_name[0] == '.')
```

Comment:
Maybe should check `.conf` suffix too, to avoid autosave files, etc. Might wanna stat and check for dirs too?

Reply:
Added ".conf" check